### DISCUSSION TITLE
How come this is a version conflict?

### DISCUSSION THREAD

**Discussion by voroninp** [NuGet/Home#13947] [Q&A]
Since when a combination of `(>= 8.0.0)` and `(>= 8.0.0 && < 9.0.0)` is a version conflict?!

```
Version conflict detected for Microsoft.Extensions.Hosting.Abstractions.
Install/reference Microsoft.Extensions.Hosting.Abstractions 9.0.0 directly to project Mbs.MediaFlowService.WebApp.Tests to resolve this issue.
 
Mbs.MediaFlowService.WebApp.Tests -> Mbs.MediaFlowService.WebApp -> Mbs.MediaFlowService.WebApp.Infrastructure -> Serilog.AspNetCore 8.0.3 -> Serilog.Extensions.Hosting 8.0.0 -> Microsoft.Extensions.Hosting.Abstractions (>= 8.0.0) 

Mbs.MediaFlowService.WebApp.Tests -> Mbs.MediaFlowService.WebApp -> Mbs.MediaFlowService.Core.Module -> Microsoft.Extensions.Hosting.Abstractions (>= 8.0.0 && < 9.0.0).
```

Here's how I reference packages:
```
<PackageVersion Include="Serilog" Version="[*, 5.0.0)" />
<PackageVersion Include="Serilog.AspNetCore" Version="[8.*, 9.0.0)" />
<PackageVersion Include="Serilog.Enrichers.Span" Version="[*, 4.0.0)" />
<PackageVersion Include="Serilog.Exceptions" Version="[8.*, 9.0.0)" />
<PackageVersion Include="Serilog.Exceptions.EntityFrameworkCore" Version="[8.*, 9.0.0)" />
<PackageVersion Include="Serilog.Sinks.Seq" Version="[8.*, 9.0.0)" />
<PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="[8.*, 9.0.0)" />
```

**Comment by voroninp** [comment]
Ok, NuGet error message is just wrong.

Another dependency (OpenTelemetry.Extensions.Hosting 1.10.0) pulls `Microsoft.Extensions.Hosting.Abstractions 9.0.0`. This satisfies constraint `>=8.0.0`, but fails for `>= 8.0.0 && < 9.0.0` which also wrong because resolved version is `8.0.1`.

P.S. I believe updating the major version of dependency should lead to the update of the major version of the package as well.

**Comment by jeffkl** [comment]
@voroninp would it be possible for you to create a simple repro of this?  I want to investigate further but just using the package references you mentioned didn't reproduce the problem.  If possible, attach a ZIP with a project and the packages that encounters this problem.
